### PR TITLE
[buteo-sync-plugins-social] Use own logging categories instead of Buteo LOG_* macros. JB#53712

### PR DESCRIPTION
**Pull request by blammit**
Add logging categories: lcSyncMLPlugin, lcSyncMLPluginTrace,
lcSyncMLContactChange, lcSyncMLContactChangeTrace.

### CODE REVIEW

File: storageplugins/hcalendar/CalendarBackend.cpp [hunk 1]
```diff
@@ -415,30 +415,30 @@ bool CalendarBackend::modifyIncidence( KCalendarCore::Incidence::Ptr aInci, cons
...
+        qCDebug(lcSyncMLPlugin) << "Single incidence committed";
     }
 
     return true;
 }
 
 CalendarBackend::ErrorStatus CalendarBackend::deleteIncidence( const QString& aUID )
 {
-	FUNCTION_CALL_TRACE;
+	FUNCTION_CALL_TRACE(lcSyncMLPluginTrace);
```
Comment (chriadam) [inline storageplugins/hcalendar/CalendarBackend.cpp]: might as well fix the indentation while we're here, I suppose

File: syncmlcommon/SyncMLCommon.h [hunk 2]
```diff
@@ -90,4 +92,8 @@ const QString SYNCML12                = "SyncML12";
 
 const QString PROF_HTTP_XHEADERS      = "http_xheaders";
 
+
+Q_DECLARE_LOGGING_CATEGORY(lcSyncMLPlugin)
```
Comment (chriadam) [inline syncmlcommon/SyncMLCommon.h:96]: should this be lcButeoSyncMLPlugin or?  I guess it doesn't matter

File: storageplugins/hcalendar/CalendarBackend.cpp (hunk 1 above)
Comment (blammit) [reply storageplugins/hcalendar/CalendarBackend.cpp]: Done

File: syncmlcommon/SyncMLCommon.h (hunk 2 above)
Comment (blammit) [reply syncmlcommon/SyncMLCommon.h:96]: lcSyncMLPlugin is probably sufficient since it's only used within this repo.

**Review by chriadam — APPROVED** [review]

**Pull request merged**